Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/LeaderBoard.hs

```diff
@@ -31,17 +31,26 @@ import Kernel.Types.Common
 import Kernel.Types.Error
 import Kernel.Types.Id
 import Kernel.Utils.Error
+import Kernel.Utils.Logging (logInfo)
 import Storage.CachedQueries.CacheConfig (CacheFlow)
 import Storage.CachedQueries.Merchant.LeaderBoardConfig as QLeaderConfig
 import qualified Storage.Queries.Person as QPerson
 
-data DriversInfo = DriversInfo
-  { rank :: Integer,
-    name :: Text,
-    totalRides :: Int,
-    totalDistance :: Meters,
-    isCurrentDriver :: Bool
-  }
+data DriversInfo
+  = RideInfo
+      { rank :: Integer,
+        name :: Text,
+        totalRides :: Int,
+        totalDistance :: Meters,
+        isCurrentDriver :: Bool
+      }
+  | ReferralInfo
+      { rank :: Integer,
+        name :: Text,
+        totalValidCustomers :: Int,
+        totalReferrals :: Int,
+        isCurrentDriver :: Bool
+      }
```

Comment:
in the above  code you can remove redundancy something like this

data Temp = Temp {
           rank :: Integer,
           name :: Text,
           isCurrentDriver :: Bool
}


then  use this 

data DriversInfo
  = RideInfo
      {
       temp :: Temp
        totalRides :: Int,
        totalDistance :: Meters,
      }
  | ReferralInfo
      { 
        temp :: Temp,
        totalValidCustomers :: Int,
        totalReferrals :: Int
      }

Reply:
Keeping backward compatibility in mind, the two responses are separated and the First response is not modified